### PR TITLE
[community extension] Add (freemarker) template-editor community extension

### PR DESCRIPTION
**Pull request by jeanpommier**
Add a (freemarker) template-editor extension

Editing freemarker templates usually requires access to disk space and some privilege escalation, which is often not very convenient. It should be possible to edit them using the web UI alongside the layers.

This extension allows to edit them in the web UI. It supports the various levels at which a freemarker template can be defined: default is from the lib, then at workspace level, store level and finally at the layer level.

For now, it only supports the freemarker templates for the WMS GetFeatureInfo response.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

### CODE REVIEW

File: src/community/template-editor/src/main/resources/GeoServerApplication.properties [hunk 1]
```diff
@@ -0,0 +1,31 @@
+TemplateEditorPage.page.title=Hello
```
Comment (mprins) [inline src/community/template-editor/src/main/resources/GeoServerApplication.properties:1]: Some of these probably need a little more work or maybe can be removed?

Comment (jeanpommier) [reply src/community/template-editor/src/main/resources/GeoServerApplication.properties:1]: woops. Can probably be removed. I'll check. Thanks for spotting this out

**Comment by aaime** [comment]
@jeanpommier you should probably rebase everything on the latest main branch, some of the failures are due to a mis-alignment between new functionality provided by GeoTools and expectations in (a few days) old GeoServer tests.

**Comment by jeanpommier** [comment]
> @jeanpommier you should probably rebase everything on the latest main branch, some of the failures are due to a mis-alignment between new functionality provided by GeoTools and expectations in (a few days) old GeoServer tests.

Oh, OK, right. Thanks @aaime . I was getting a bit desperate

**Comment by jeanpommier** [comment]
> @jeanpommier you should probably rebase everything on the latest main branch, some of the failures are due to a mis-alignment between new functionality provided by GeoTools and expectations in (a few days) old GeoServer tests.

@aaime it was not jut that a rebase was needed, but I believe now it is OK. At least, it seems to preperly build without errors

File: doc/en/user/source/community/template-editor/index.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+#. Extract the contents of the archive into the ``WEB-INF/lib`` directory of the GeoServer installation.
+
+#. Reload GeoServer.
+
+
+Accessing the Template Editor
+-----------------------------
+
+Log in your GeoServer web UI. If the extension has been properly installed, a *Template Edition* entry should appear on the left menu, Data section, below *Styles*. You can also check in the list of Modules, in the *Server Status* menu entry.
```
[the remark this answers is not in the document]
Comment (jeanpommier) [reply doc/en/user/source/community/template-editor/index.rst:28]: Thanks @aaime  for the review. Sorry for the late reply, I was on holiday. I'll look at this when I come back from foss4g.

[the remark this answers is not in the document]
Comment (jeanpommier) [reply doc/en/user/source/community/template-editor/index.rst:28]: Thanks @aaime , I'll do that as soon as I get a moment to do it. 

I'm really not comfortable with Wicket, but following your hint, I can see 2 possible sources for this behaviour:
- the templateeditor pages all inherit from an [abstract page class](https://github.com/pi-geosolutions/geoserver/blob/template-editor-extension-main/src/community/template-editor/src/main/java/org/geoserver/template/editor/web/AbstractTemplateEditorPage.java)
- On the editor page, I am implementing 3 instances of the same [TemplateFormPanel](https://github.com/pi-geosolutions/geoserver/blob/template-editor-extension-main/src/community/template-editor/src/main/java/org/geoserver/template/editor/web/TemplateFormPanel.java), one the the header, another one for the content, and the last one for the footer.

Does this ring a bell for you ?

**Pull request closed without merging**